solr/core/src/java/org/apache/solr/search/QueryResultKey.java

```diff
@@ -51,20 +52,28 @@ public QueryResultKey(
       Query query, List<Query> filters, Sort sort, int nc_flags, int minExactCount) {
     this.query = query;
     this.sort = sort;
-    this.filters = filters;
     this.nc_flags = nc_flags;
     this.minExactCount = minExactCount;
 
     int h = query.hashCode();
 
-    if (filters != null) {
-      for (Query filt : filters)
+    if (filters == null) {
+      this.filters = null;
+    } else {
+      this.filters = filters.stream().filter(Objects::nonNull).collect(Collectors.toList());
+      for (Query filt : this.filters) {
```

Comment:
what about using stream instead of for loop?
> h += this.filters.stream().mapToInt(Query::hashCode).sum();

Reply:
I'm open to it...   While I like for loops, maybe cause I'm not up on stream, we already are doing that...